twenty-first/src/math/digest.rs

```diff
@@ -206,6 +217,51 @@ impl Digest {
     pub fn hash<H: AlgebraicHasher>(self) -> Digest {
         H::hash_pair(self, Digest::new([BFieldElement::zero(); DIGEST_LENGTH]))
     }
+
+    /// Encode digest as hex
+    pub fn to_hex(&self) -> String {
+        let bytes = <[u8; Digest::BYTES]>::from(*self);
+        hex::encode(bytes)
+    }
+
+    /// Decode hex string to Digest
+    pub fn try_from_hex(string: &str) -> Result<Self, FromHexDigestError> {
```

Comment:
Would it make sense to use the same trait bound as `hex::decode`?
```suggestion
    pub fn try_from_hex(data: impl AsRef<[u8]>) -> Result<Self, FromHexDigestError> {
```

Reply:
yes!  good catch, thx.   I meant to go back and do that, and it slipped my mind.  Friendlier for callers.